### PR TITLE
Implementation for rule checking if last password change is in the past

### PR DESCRIPTION
**Pull request by teacup-on-rockingchair**
#### Description:

- Add rule  checking if last password change is in the past

#### Rationale:

- Add OVAL rule to check time difference between now and last change date of shadow entry
- Add accounts_password_last_change_is_in_past rule to CIS profiles
- Add basic pass and fail test scenarios

#### Review Hints:

- There is a strange issue in OVAL rule I observed:
There is strange issue I observed that if time_difference function is passed larger number of seconds than now(i.e. time in future)
    It returns a very big integer instead of negative number so I had to monkeypatch the initial idea of the rule to check if the time difference is larger than 1000 years, then it means it was a negative result and we have a failure.
    
    For example with the test code for fail scenario, i.e. putting last change date for tomorrow I get with time_difference the result 184467440737095,
    while for dates in the past the result look ok.

- _Use this optional section to give any relevant information which could help the reviewer to more quickly and assertively understand and test the changes._

- _Good examples are useful commands, if it is better to review all commits together or in a suggested sequence, any relevant discussion in other PRs or issues, etc._

### CODE REVIEW

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml:5]: Could you adjust the OVAL so it fit in 99 columns, please?

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml:7]: Please, include an empty line between the sections.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml:8]:
```suggestion
    If a user recorded password change date is in the future then they could
```

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml:20]:
```suggestion
```

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/user_password_change_time_in_future.fail.sh
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/user_password_change_time_in_future.fail.sh:3]: nitpick: `testUserChage1` is used 3 times. Maybe a variable for it would improve the readability.

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
Thanks for this rule. I shared some comments more related to style guide.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml:26]: This is the error reported by many CI tests:
`oval:ssg-state_accounts_password_last_change_time_diff:ste:1 - a var_check has been supplied for the ind:value entity so a var_ref must also be provided`

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml]: :+1: changed in [135d663](https://github.com/ComplianceAsCode/content/pull/10031/commits/135d663c3dac8a18fb51088c0b758f095081c5fd)

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml:8]: :+1: changed in [135d663](https://github.com/ComplianceAsCode/content/pull/10031/commits/135d663c3dac8a18fb51088c0b758f095081c5fd)

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml]: :+1: changed in [135d663](https://github.com/ComplianceAsCode/content/pull/10031/commits/135d663c3dac8a18fb51088c0b758f095081c5fd)

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml:21]: :+1: changed in [135d663](https://github.com/ComplianceAsCode/content/pull/10031/commits/135d663c3dac8a18fb51088c0b758f095081c5fd)

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/user_password_change_time_in_future.fail.sh
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/user_password_change_time_in_future.fail.sh]: :+1: changed in [135d663](https://github.com/ComplianceAsCode/content/pull/10031/commits/135d663c3dac8a18fb51088c0b758f095081c5fd)

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml:30]: Dropped that thing in [135d663](https://github.com/ComplianceAsCode/content/pull/10031/commits/135d663c3dac8a18fb51088c0b758f095081c5fd) , probably remained from previous version of how the state section was written. Thanks :bow:

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml:33]: Although a remediation for this rule would be possible to be automated by Bash and Ansible, I totally agree to not do that in order to avoid any system disruption. However, it would be great to include a [`warning`](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#rule-sections) informing the reason there is no automated remediation for that. This way user can see the warning in the guide.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml:38]: Would the intention to express `less than or equal today`?

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml]: :+1:  added more descriptive fixtext and warning for automatic remediation absence in b59dece

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/oval/shared.xml]: definitely true :) changed it to more accurate comment in 151e230ef4b0662dcf42513c09bebc2373eab16b

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml:35]: `${lines[@]}` is not defined.

Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml:35]: `${lines[@]}` is not defined.

Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml]: I guess I parsed the script without the first line :facepalm:

**Comment by marcusburghardt** [comment]
Automatus CS8, CS9 and Fedora are failing because the containers used for test don't have any unlocked user with a password set. The test scenario could be extended to manually include a valid line in /etc/shadow, but I don't think it is necessary. The tests are behaving as expected in VMs.

@teacup-on-rockingchair , I am fine to approve and merge it unless you prefer to update the test scenario to not fail in containers. Its on your call. Only let me know. Thanks

**Comment by marcusburghardt** [comment]
> > Automatus CS8, CS9 and Fedora are failing because the containers used for test don't have any unlocked user with a password set. The test scenario could be extended to manually include a valid line in /etc/shadow, but I don't think it is necessary. The tests are behaving as expected in VMs.
> > @teacup-on-rockingchair , I am fine to approve and merge it unless you prefer to update the test scenario to not fail in containers. Its on your call. Only let me know. Thanks
> 
> In general I would prefer to have my tests valid for all environments, so I would prefer to add a test user to the pass scenario also so all is as green as possible ;)

Great!! In that case, or the test scenario script should ensure the `passwd` package by defining a header. Or a line should be manually included in `/etc/shadow`.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/all_password_change_time_in_past.pass.sh
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/all_password_change_time_in_past.pass.sh:2]:
```suggestion
```
Now we can remove this header since the `passwd` command is no longer used.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/user_password_change_time_in_future.fail.sh
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/user_password_change_time_in_future.fail.sh:3]:
```suggestion
```
Now we can remove this header since the `passwd` command is no longer used.

Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/user_password_change_time_in_future.fail.sh]: 👍

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/all_password_change_time_in_past.pass.sh
Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/all_password_change_time_in_past.pass.sh:2]: @teacup-on-rockingchair , could you remove here too, please? This should save some seconds during the test scenarios when this package is not installed.

Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/tests/all_password_change_time_in_past.pass.sh:2]: I am a bad clicker obviously 🤦

**Review by marcusburghardt — APPROVED** [review]
Thanks for including this rule.

**Comment by marcusburghardt** [comment]
All tests have passed in local VMs.
Overriding the CODEOWNERS protection since there is no SUSE contributor with write permission at the moment.

**Pull request merged**
